src/service/resources-impl.js

```diff
@@ -1353,8 +1353,8 @@ export class Resources {
     }
 
     if (request) {
-      request.newHeight = newHeight;
-      request.newWidth = newWidth;
+      request.newHeight = newHeight || request.newHeight;
+      request.newWidth = newWidth || request.newWidth;
```

Comment:
This avoids two `resources#changeSize` methods from clobbering each other. Without this change:
```
changeSize(el, 100, undefined)
changeSize(el, undefined, 100)
```
`el` won't have its height changed to 100 since the second call's `newHeight` value of `undefined` will override the first call.

Perhaps I should move this to a separate PR since it's a somewhat scary change?